eng/pipelines/coreclr/templates/build-job.yml

```diff
@@ -164,8 +166,8 @@ jobs:
         continueOnError: true
         condition: always()
 
-    # We only test on clang binaries.
-    - ${{ if eq(parameters.compilerName, 'clang') }}:
+    # Builds using gcc are not tested.
+    - ${{ if ne(parameters.compilerName, 'gcc') }}:
```

Comment:
In fact, I suspect the motivation for the conditional statement was that, at the time gcc compilation support was being introduced, it was decided to only do builds without testing to avoid stressing the lab capacity too much and perhaps that's why publishing of product artifacts was suppressed for the gcc case. It could be also due to the fact I described in the other comment - if we were doing both flavors of CoreCLR compilation in the same pipeline, the two artifacts would clash. In fact it would be probably clearer to include the compiler in the names but it's obviously more work as the dependencies reside in multiple scripts.

Reply:
[In other words I think we might be able to just delete this conditional clause. Product publishing takes a few seconds and for the few GCC runs that do it unnecessarily it shouldn't be a big concern and it would at least produce some verifiable artifacts in Azure.]

Reply:
I don't think we should remove this clause. We shouldn't publish gcc artifacts to avoid clashes, and specially if we don't need them afterwards to save bandwidth when uploading and such.